tensorflow/compiler/xla/service/hlo_instruction.cc

```diff
@@ -3190,6 +3204,12 @@ StatusOr<HloInstruction::FusionKind> StringToFusionKind(
   return InvalidArgument("Unknown fusion kind: %s", kind_name);
 }
 
+string FrontendAttributesToString(
+    const FrontendAttributes& frontend_attributes) {
+  return absl::StrFormat("{%s}", absl::StrJoin(frontend_attributes.map(), ",",
```

Comment:
I don't think we can rely on a stable iteration order for `frontend_attributes.map()`, so this will give us non-deterministic serialization of HLO instructions.

Reply:
Done